core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManager.java

```diff
@@ -452,7 +453,9 @@ public InputStream fetchLogSegment(final RemoteLogSegmentMetadata remoteLogSegme
 
             final var suffix = ObjectKeyFactory.Suffix.LOG;
             final var segmentKey = objectKey(remoteLogSegmentMetadata, suffix);
-
+            if (chunkManager instanceof ChunkCache) {
+                ((ChunkCache<?>) chunkManager).startPrefetching(segmentKey, segmentManifest, range.from);
+            }
```

Comment:
Any reason this is happening here and not in `ChunkCache#getChunk`? Looks like a leaky abstraction.

Reply:
You are right, fixed.